### PR TITLE
Community resources

### PR DESCRIPTION
**Pull request by ultimatemonty**
Adds endpoints for community resources.

/cc @pdougall1 @camthompson @gschorkopf

### CODE REVIEW

**Comment by gschorkopf** [comment]
I might say that the commit history is a lil wonky. I'm going to review this based on file changes, but ideally the commits should tell a clear story chronologically.

File: Gemfile [hunk 1]
```diff
@@ -8,6 +8,10 @@ gem 'rails', '4.1.9'
 # use postgres database
 gem 'pg'
 
+group :development, :test do
+  gem 'sqlite3'
```
Comment (gschorkopf) [inline Gemfile:12]: Why not use postgres everywhere? I don't think it's worth adding another dependency.

File: app/controllers/community_resources_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+class CommunityResourcesController < ApplicationController
+  before_filter :set_resource, :except => [:index, :create]
```
Comment (gschorkopf) [inline app/controllers/community_resources_controller.rb:2]: Ruby 1.9.2 and above + new Rails above use this syntax:

``` ruby
before_action :set_resource, except: [:index, :create]
```

File: app/controllers/community_resources_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  def set_resource
+    @resource = CommunityResource.find(params['id'])
```
Comment (gschorkopf) [inline app/controllers/community_resources_controller.rb:4]: `params[:id]`. ^ will work, but it's uncommon.

File: app/helpers/community_resource_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module CommunityResourceHelper
```
Comment (gschorkopf) [inline app/helpers/community_resource_helper.rb:1]: I'd remove this if it's not used.

File: db/schema.rb [hunk 5]
```diff
@@ -23,6 +23,19 @@
     t.datetime "updated_at"
   end
 
+  create_table "community_resources", force: true do |t|
```
Comment (gschorkopf) [inline db/schema.rb:26]: This is a nitpick, but I think `Resource` is probably a sufficient name.

File: spec/controllers/community_resource_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require 'spec_helper'
+
+describe CommunityResourceController do
```
Comment (gschorkopf) [inline spec/controllers/community_resource_controller_spec.rb]: I'd toss this file too.

File: spec/helpers/community_resource_helper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+# the CommunityResourceHelper. For example:
+#
+# describe CommunityResourceHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+describe CommunityResourceHelper do
```
Comment (gschorkopf) [inline spec/helpers/community_resource_helper_spec.rb]: (And this)

File: spec/models/community_resource_spec.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+require 'spec_helper'
+
+describe CommunityResource do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (gschorkopf) [inline spec/models/community_resource_spec.rb]: (And this)

File: Gemfile (hunk 1 above)
Comment (ultimatemonty) [inline Gemfile:12]: @gschorkopf I didn't have postgres setup when @camthompson and I initially worked on this so we used sqlite as a workaround. Will remove

File: app/controllers/community_resources_controller.rb (hunk 2 above)
Comment (ultimatemonty) [inline app/controllers/community_resources_controller.rb:2]: @gschorkopf found that in a code sample online. Will update to the newer syntax.

On a side note - do you know any good online tutorials around creating an API with Rails? I found a fair amount of stuff but most of it was older (2014) and wasn't sure if they used current best practices.

File: db/schema.rb (hunk 5 above)
Comment (ultimatemonty) [inline db/schema.rb:26]: @gschorkopf I felt that `resource` was too generic of a name and opted for `community_resource` to be a bit more explicit.

**Comment by ultimatemonty** [comment]
@gschorkopf can `atlanta-ember-backend/atlanta_ember_backend_development` and `atlanta-ember-backend/atlanta_ember_backend_test` be removed/ignored? They're binaries and I'm sure wtf they do.

**Comment by gschorkopf** [comment]
If those are just your sqlite databases, then yeah.

**Comment by gschorkopf** [comment]
@ultimatemonty I'm going to take a stab at fixing this guy up. I'd like to get this in in good shape before tackling JSONAPI.org stuff

**Comment by gschorkopf** [comment]
The test suite won't run. Issue with `uninitialized constant CommunityResourceController (NameError)`. The bug here is just naming: should be plural, `CommunityResourcesController`. But #17 would help us figure this out sooner.

Edit: Turns out either this branch (or develop/master) is failing.

**Comment by gschorkopf** [comment]
Tried to fix a few specs. But now I see that master is passing these; except for `meetup_spec.rb` :frowning:

**Comment by ultimatemonty** [comment]
@gschorkopf talked with @pdougall1 and @camthompson about this last night. Going to close this PR and not merge in favor of the Community Resources page being a easy way for folks to learn how to PR and contribute to a project. Thanks for all the feedback - it was good stuff!

I'll have a new PR later this week on the client that will add a note something to the effect of "Have a great resource you want to share? Submit a PR!"

**Pull request closed without merging**